xbmc/music/MusicLibraryQueue.cpp

```diff
@@ -70,15 +70,14 @@ void CMusicLibraryQueue::ExportLibrary(const CLibExportSettings& settings, bool
   if (showDialog)
   {    
     AddJob(exportJob);
+    m_done.Reset();
 
     if (progress)
     {
-      // Render and wait for export to complete or be cancelled
-      while (progress->IsActive() && !progress->IsCanceled())
+      // Render (so pointer movements work on dialog) and wait for export to 
+      // complete or be canceled
+      while (!m_done.WaitMSec(10) && progress->IsActive() && !progress->IsCanceled())
```

Comment:
Is specifying a wait time here necessary? The wait could be unconditional if changing to active and cancel states sets the event.

Reply:
To make the mouse move smoothly it needs the dialog to render, hence calling Progress(). The wait is to trigger that every 10ms, sufficent to make mouse movement smooth, rather than at processor speed.